src/test/java/com/google/sps/tests/NewsTest.java

```diff
@@ -0,0 +1,152 @@
+  */
+  private static String textToString(String filePath) throws IOException {
+    StringBuilder stringBuilder = new StringBuilder();
+    try (BufferedReader inputReader = new BufferedReader(new FileReader(filePath))) {
+      String inputLine;
+      while ((inputLine = inputReader.readLine()) != null) {
+        stringBuilder.append(inputLine);
+      }
+    } catch(IOException e) {
+      System.err.println(e);
+      return "";
+    }
+    return stringBuilder.toString();
+  }
+
+ /**
+  * Helper method to check the makeArticleList method in the NewsServlet class.
+  * Checks if makeArticleList on HTML text in the specified file has output equivalent to expectedList.
+  */
+  private void checkMakeArticleList(String filePath, List<Article> expectedList) throws IOException {
+    String htmlString = textToString(filePath);
+    List<Article> articleList = ns.makeArticleList(htmlString);
+    // Check each class instance in each article object is equal.
+    Assert.assertEquals(articleList.size(), expectedList.size());
+    for(int i = 0; i < articleList.size(); i++) {
+      Article realArticle = articleList.get(i);
+      Article expectedArticle = expectedList.get(i);  
+      Assert.assertEquals(expectedArticle.getTitle(), realArticle.getTitle());
+      Assert.assertEquals(expectedArticle.getLink(), realArticle.getLink());
+    }  
+  }
+  
+ /**
+  * Set up for following tests.
+  * Populates someArticleList with Article entries.
+  */
+  @BeforeClass
+  public static void setUp() {
+    someArticleList.add(new Article(
+      "The latest Instagram trend: putting your name on photos of frogs, cats and Harry Styles",
+      "news.google.com/articles/CBMib2h0dHBzOi8vd3d3LnRoZWd1YXJkaWFuLmNvbS91cy1uZXdzLzIwMjAvanVsLzIwL2luc3RhZ3JhbS13aGF0LWZyb2ctYXJlLXlvdS1pbnNpZGUtdGhlLWxhdGVzdC13YXktdG8td2FzdGUtdGltZdIBb2h0dHBzOi8vYW1wLnRoZWd1YXJkaWFuLmNvbS91cy1uZXdzLzIwMjAvanVsLzIwL2luc3RhZ3JhbS13aGF0LWZyb2ctYXJlLXlvdS1pbnNpZGUtdGhlLWxhdGVzdC13YXktdG8td2FzdGUtdGltZQ?hl=en-US&amp;gl=US&amp;ceid=US%3Aen"
+    ));
+    someArticleList.add(new Article(
+      "How a blue protein turns tree frogs bright green",
+      "news.google.com/articles/CBMiVmh0dHBzOi8vd3d3LnNjaWVuY2VtYWcub3JnL25ld3MvMjAyMC8wNy9ob3ctYmx1ZS1wcm90ZWluLXR1cm5zLXRyZWUtZnJvZ3MtYnJpZ2h0LWdyZWVu0gEA?hl=en-US&amp;gl=US&amp;ceid=US%3Aen"
+    ));
+  }
+
+  @Before
+  public void createNewsServlet() throws IOException {
+    System.out.println("createNewsServlet()");
+    ns = new NewsServlet();
+  }
+
+ /**
+  * Check that when given no HTML text, makeArticleList returns an empty list.
+  */
+  @Test
+  public void articlesFromEmptyHTML() throws IOException {
+    List<Article> articleList = ns.makeArticleList("");
+    Assert.assertEquals(articleList, new ArrayList<Article>());
+  }
+
+ /**
+  * Checks that makeArticleList makes a list with all the articles in the HTML text
+  * if there are less articles in the HTML text than makeArticleList would genereate by default.
```

Comment:
[Optional] Line wrapping looks a bit weird here.  Can't you find more words on the first line w/o going over the 80 character limit?